src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs

```diff
@@ -238,6 +239,8 @@ private void ChangeProjectProperty<T>(
         Func<SolutionChangeAccumulator, ProjectUpdateState, T, ProjectUpdateState> updateSolution,
         bool logThrowAwayTelemetry = false)
     {
+        using var _ = CreateBatchScope();
```

Comment:
Looks like CreateBatchScope takes the semaphore inside.  Any benefit of creating a `CreateBatchScope_NoLock` and moving it inside the gate we already take below?

Reply:
Talked offline.  Def an option.  Would prefer to see if we actually run into any contention that warrants this.